x-pack/plugins/observability/public/hooks/use_get_filtered_rule_types.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { usePluginContext } from './use_plugin_context';
+
+export function useGetFilteredRuleTypes() {
```

Comment:
This particular pattern was used in both the Rules and the Rule Detail page, so it made sense to turn them into their own hook.